### ISSUE TITLE
Durandal: where does it reside on the radar?

### ISSUE THREAD

**Issue by deneboulton** [red-gate/internal-tech-radar#29]
We [COREDev] have been using Durandal now for a couple of months - it is used within the 'Accounts/My Products' section of the dynamic red.gate.com website.

Since react.js is on the radar it seems that Durandal should be on it somewhere too?
I believe Gearset, DLM Dashboard and SQL Clone (though soon to be react.js) also use Durandal.

We started to use it because we thought it would be a good choice for our "experiment" with the licensing portal as there was already knowledge of it in the building.

Thoughts?

**Comment by fffej** [comment]
Where'd you think it should be? 

My guess would be that we've used it in Gearset, DLM Dashboard and Clone (though be interested to hear from @GrahamTheCoder and team about why it might be react.js in the future) then it should be in the "validate" section.

**Comment by adrianbanks** [comment]
Isn't [Aurelia](http://aurelia.io) supposed to be the next step for Durandal?

**Comment by GrahamTheCoder** [comment]
My hunch is that React is the thing every new project should use, I might be slightly jumping the gun.
It's worth pointing out that SQL Clone is using other stuff on top of it since it's pretty lightweight. So we're using Redux and Redux-form on top of that. See lightning talks for a few more details.

Thoughts @thomashodgson, @allymparker, @tugberkugurlu?

**Comment by deneboulton** [comment]
@fffej Given that at least COREDEV are currently using it in a user facing application it feels like it should be in the 'validate' section - but we should do that validation ASAP and then decide if it should be updated to 'JFDI' or 'no more of this'.
I suspect there is enough experience in RG with Durandal that this validation could happen now, unless it depends on whether React.js gets promoted to JFDI?

I say ASAP because the licensing portal will likely grow in functionality and could feasibly "expand" out and be the go to place for more customer workflows on the Redgate website.

@adrianbanks I would say that Aurelia is more of a spiritual successor to Durandal.

**Comment by fffej** [comment]
I'm worried about making any web-framework JFDI, since http://dayssincelastjavascriptframework.com/ is always 0. More seriously, the web is still evolving at breakneck speed - I think we need to accept that and use the technology that's most appropriate. I'd be happy with this in Validate but I'd need more convincing on the JFDI.

**Comment by alastairs** [comment]
> @adrianbanks I would say that Aurelia is more of a spiritual successor to Durandal.

Yeah, I'd agree with that. Aurelia is much closer to AngularJS (2.0) in design and philosophy than Durandal, which is something I'm a bit uncomfortable with, tbh. Durandal's philosophy is one of "pull together all these awesome bits for you and provide little additional functionality", whereas Aurelia's feels more like "you will be an Aurelia app". (FWIW, React feels like it sits somewhere in the middle for me.)

**Comment by deneboulton** [comment]
@fffej Just to be clear; I am not necessarily pushing for Durandal to reside in the JFDI section, I am asking the question and pushing for validation so that we do not invest too much time and effort into a framework that is categorised as 'no more of this'.

Personally, the main thing that I want from a tech-radar is what not to do.
At what point can we say something has been validated? Is it dependent on having enough strong opinion?

**Comment by GrahamTheCoder** [comment]
Ah, I thought @fffej asked about React seeing them as competitors where one should be the recommendation for new projects. I have since realized that it would be possible to use both as someone implied in a [blog I read about 8 words of](http://ilikekillnerds.com/2015/03/aurelia-vs-react-js-based-on-actual-use/).

Regardless, I'd put React in the category above Durandal just for its popularity in the world right now

**Comment by tugberkugurlu** [comment]
> I'm worried about making any web-framework JFDI, since http://dayssincelastjavascriptframework.com/ is always 0. 

I agree with that statement. It changes quite drastically. However, I also agree with the "[Pick a Horse and Ride](https://wildermuth.com/2015/09/01/Angular_v_React_v_Aurelia_v_et_al)" metaphor on JS libraries a bit. So, picking one and going with that is quite valuable. [React cares about stability](https://facebook.github.io/react/contributing/design-principles.html#stability) too much which is a good thing in a fast evolving ecosystem.

However, one important thing to keep in mind when comparing React to Durandal, Aurelia, etc. is that React is actually a library to write UI components whereas others are full-blown frameworks which has routing, etc. in them.

**Comment by fffej** [comment]
@tugberkugurlu makes the point "to pick a horse and ride it". If now is the time to do that, then it seems React is the library for building components and bets are open for frameworks. If there's appetite to get more solid decisions I'd like #guild-web to get involved as that's the group with the expertise in this area.

**Comment by tugberkugurlu** [comment]
> building components and bets are open for frameworks

I think that's a sensible approach. There 2 few sides of the picture here:

 1. You can build your whole app with React without using any other heavy frameworks. Libraries like [react-router](https://github.com/reactjs/react-router), [redux](https://github.com/reactjs/redux), [react-redux](https://github.com/reactjs/react-redux), [redux-form](https://github.com/erikras/redux-form) compliment React in order to achieve this easily.

 2. You can use React to build your reusable ([optionally dumb](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0#.yssccco4z)) components to solve the UI problems and use another full-blown framework to handle the life-cycle of the app. For instance, Aurelia can work with React: https://github.com/bryanrsmith/aurelia-react-loader and load React component directly.
